prow/cmd/branchprotector/protect.go

```diff
@@ -289,3 +310,34 @@ func (p *protector) UpdateBranch(orgName, repo string, branchName string, branch
 	}
 	return nil
 }
+
+func equalBranchProtections(bp1, bp2 github.BranchProtection) bool {
+	if bp1.RequiredStatusChecks != nil {
+		sort.Strings(bp1.RequiredStatusChecks.Contexts)
+	}
+	if bp2.RequiredStatusChecks != nil {
+		sort.Strings(bp2.RequiredStatusChecks.Contexts)
+	}
+
+	rp1 := bp1.RequiredPullRequestReviews
+	rp2 := bp2.RequiredPullRequestReviews
+	if rp1 != nil {
+		sort.Strings(*rp1.DismissalRestrictions.Users)
+		sort.Strings(*rp1.DismissalRestrictions.Teams)
+	}
+	if rp2 != nil {
+		sort.Strings(*rp2.DismissalRestrictions.Users)
+		sort.Strings(*rp2.DismissalRestrictions.Teams)
+	}
+
+	if bp1.Restrictions != nil {
+		sort.Strings(*bp1.Restrictions.Teams)
+		sort.Strings(*bp1.Restrictions.Users)
+	}
+	if bp2.Restrictions != nil {
+		sort.Strings(*bp2.Restrictions.Teams)
+		sort.Strings(*bp2.Restrictions.Users)
+	}
```

Comment:
nit: Instead of duplicating all the sorting logic, you could move it to an anonymous function. e.g.
```go
sortBP := func(bp *github.BranchProtection) {
  // sort bp
}
sortBP(&bp1)
sortBP(&bp2)
```

Reply:
Thank you for the code review. This reads much better. I will update the PR.

Reply:
Since this was asked as a question, I have left as-is pending a decision form maintainers. Please let me know if the casing of strings needs to be considered as well.